### PR TITLE
[docs] Add a Kagami note

### PR DESCRIPTION
**Pull request by 6r1d**
I'd like to add a note on `kagami` tool to the account registration description.
Related issue: https://github.com/hyperledger/iroha-2-docs/issues/82

Signed-off-by: 6r1d <vic.6r1d@gmail.com>

### CODE REVIEW

File: src/guide/bash.md [hunk 1]
```diff
@@ -241,14 +241,51 @@ With a domain available, it is time to register an account.
...
+The `--id` argument above specifies the **account name** and the
+**domain**, while `--key` argument specifies the
+[public key](https://en.wikipedia.org/wiki/Public-key_cryptography).
+
+It is possible to generate both `public` and `private` key using an
+internal tool, `kagami`. One can either run `kagami` using Cargo or find it
+in the build directories; for the `debug` build, kagami will be located in
+`target/debug/` directory.
+
+To generate a new key pair, run it without arguments.
```
Comment (outoftardis) [inline src/guide/bash.md:259]:
```suggestion
To generate a new key pair, run it without any arguments.
```

File: src/guide/bash.md [hunk 2]
```diff
@@ -241,14 +241,51 @@ With a domain available, it is time to register an account.
...
+
+```bash
+# "cargo run" syntax
+# cargo run --bin kagami crypto
+$ ./kagami crypto
+Public key (multihash): ed012088a5d0cada6edfd4ad8046da64ffce820c16f0e68a21f98df505becd5011e18a
+Private key: 749682bd40695cfc77b5ad599220a80d69b72db5e5078c259635f8e8d8f3559088a5d0cada6edfd4ad8046da64ffce820c16f0e68a21f98df505becd5011e18a
+Digest function: ed25519
```
Comment (appetrosyan) [inline src/guide/bash.md:270]:
```suggestion

To use `cargo`, run 
```bash
$ cargo run --bin kagami  -- crypto
Public key (multihash): ed012088a5d0cada6edfd4ad8046da64ffce820c16f0e68a21f98df505becd5011e18a
Private key: 749682bd40695cfc77b5ad599220a80d69b72db5e5078c259635f8e8d8f3559088a5d0cada6edfd4ad8046da64ffce820c16f0e68a21f98df505becd5011e18a
Digest function: ed25519
```

If you want to copy the program somewhere convenient `e.g. /bin` for system-wide use you need to build it first
```bash
$ cargo build --bin kagami 
$ cd target/debug
```
Move/copy to the destination and from the destination folder run with 
```bash
$ ./kagami crypto
Public key (multihash): ed012088a5d0cada6edfd4ad8046da64ffce820c16f0e68a21f98df505becd5011e18a
Private key: 749682bd40695cfc77b5ad599220a80d69b72db5e5078c259635f8e8d8f3559088a5d0cada6edfd4ad8046da64ffce820c16f0e68a21f98df505becd5011e18a
Digest function: ed25519
```

File: src/guide/bash.md [hunk 3]
```diff
@@ -241,14 +241,51 @@ With a domain available, it is time to register an account.
...
+pair with `-a` flag and seed with `-s` flag:
+
+```bash
+# "cargo run" syntax
+# cargo run --bin kagami -- crypto --algorithm ed25519 --seed 42
+$ ./target/debug/kagami crypto -a ed25519 -s 42
+Public key (multihash): ed0120ee7a84e8ed0ecc5ae65968ff5465d2bc16f5c6211bc4de0a3f1591e82845f824
+Private key: 790e18703ba023439ee28be38def744ae613cd5e90c643434571b6de80666123ee7a84e8ed0ecc5ae65968ff5465d2bc16f5c6211bc4de0a3f1591e82845f824
+Digest function: ed25519
+```
```
Comment (appetrosyan) [inline src/guide/bash.md:283]:
```suggestion
```bash
# "cargo run" syntax
cargo run --bin kagami -- crypto --algorithm ed25519 --seed 42
Public key (multihash): ed0120ee7a84e8ed0ecc5ae65968ff5465d2bc16f5c6211bc4de0a3f1591e82845f824
Private key: 790e18703ba023439ee28be38def744ae613cd5e90c643434571b6de80666123ee7a84e8ed0ecc5ae65968ff5465d2bc16f5c6211bc4de0a3f1591e82845f824
Digest function: ed25519
```
```

File: src/guide/bash.md [hunk 4]
```diff
@@ -241,14 +241,51 @@ With a domain available, it is time to register an account.
...
+# "cargo run" syntax
+# cargo run --bin kagami crypto
+$ ./kagami crypto
+Public key (multihash): ed012088a5d0cada6edfd4ad8046da64ffce820c16f0e68a21f98df505becd5011e18a
+Private key: 749682bd40695cfc77b5ad599220a80d69b72db5e5078c259635f8e8d8f3559088a5d0cada6edfd4ad8046da64ffce820c16f0e68a21f98df505becd5011e18a
+Digest function: ed25519
+```
+
+It is also possible to explicitly pick an algorithm for generating a key
+pair with `-a` flag and seed with `-s` flag:
```
Comment (appetrosyan) [inline src/guide/bash.md:274]:
```suggestion
::: info 
It is also possible to explicitly pick an algorithm for generating a key
pair with `-a` flag and seed with `-s` flag:

For example, to generate an `ed25519` key (the default algorithm). 
```

Comment (outoftardis) [inline src/guide/bash.md:271]: maybe something along the lines:
```suggestion
You can also use the `-a` flag to specify the algorithm  for key-pair generation and the `-s`
flag to specify the seed:
```

File: src/guide/bash.md [hunk 5]
```diff
@@ -241,14 +241,48 @@ With a domain available, it is time to register an account.
...
+# cargo run --bin kagami -- crypto --algorithm ed25519 --seed 42
+$ ./target/debug/kagami crypto -a ed25519 -s 42
+Public key (multihash): ed0120ee7a84e8ed0ecc5ae65968ff5465d2bc16f5c6211bc4de0a3f1591e82845f824
+Private key: 790e18703ba023439ee28be38def744ae613cd5e90c643434571b6de80666123ee7a84e8ed0ecc5ae65968ff5465d2bc16f5c6211bc4de0a3f1591e82845f824
+Digest function: ed25519
+```
+
+Save the key pair, so you can use it for authenticating in Iroha, for
+instance, by replacing the `PUBLIC_KEY` and `PRIVATE_KEY` parameters in
+`config.json` from [step 1](#_1-iroha-2-client-setup).
```
Comment (outoftardis) [inline src/guide/bash.md:284]: I don't think you need that much info here, just refer to the step 1, e.g.:

```suggestion
Save the key pair, so you can use it for authentication by
[configuring the corresponding parameters](#_1-iroha-2-client-setup).
```

File: src/guide/bash.md [hunk 6]
```diff
@@ -241,14 +241,51 @@ With a domain available, it is time to register an account.
...
 
 ```bash
 ./iroha_client_cli account register \
     --id="mad_hatter@looking_glass" \
     --key="ed0120a753146e75b910ae5e2994dc8adea9e7d87e5d53024cfa310ce992f17106f92c"
 ```
 
+The `--id` argument above specifies the **account name** and the
+**domain**, while `--key` argument specifies the
+[public key](https://en.wikipedia.org/wiki/Public-key_cryptography).
```
[the remark this answers is not in the document]
Comment (6r1d) [reply src/guide/bash.md]: Absolutely, thanks!

File: src/guide/bash.md [hunk 7]
```diff
@@ -241,14 +241,51 @@ With a domain available, it is time to register an account.
...
+
+It is possible to generate both `public` and `private` key using an
+internal tool, `kagami`. One can either run `kagami` using Cargo or find it
```
[the remark this answers is not in the document]
Comment (6r1d) [reply src/guide/bash.md]: Can't apply now

File: src/guide/bash.md [hunk 8]
```diff
@@ -241,14 +241,51 @@ With a domain available, it is time to register an account.
...
+in the build directories; for the `debug` build, kagami will be located in
+`target/debug/` directory.
```
[the remark this answers is not in the document]
Comment (6r1d) [reply src/guide/bash.md]: Can't apply now either

File: src/guide/bash.md [hunk 9]
```diff
@@ -241,14 +241,48 @@ With a domain available, it is time to register an account.
...
+To register a new account called _mad_hatter_ within the _looking_glass_
+domain, run:
 
 ```bash
 ./iroha_client_cli account register \
     --id="mad_hatter@looking_glass" \
     --key="ed0120a753146e75b910ae5e2994dc8adea9e7d87e5d53024cfa310ce992f17106f92c"
 ```
 
+The `--id` argument in the above code snippet specifies the _account id_, the unique name assigned to that account in the blockchain, which includes the _domain_ which is a group of things like asset definitions, account id's and other objects that we'll talk about later. The `--key` argument specifies the
```
Comment (outoftardis) [inline src/guide/bash.md:253]:
```suggestion
The `--id` argument in the above code snippet specifies the _account id_, the unique name assigned to that account in the blockchain. This name includes the _domain_, which is a group of things like asset definitions, account ids and other objects that we'll talk about later. The `--key` argument specifies the
```

File: src/guide/bash.md [hunk 10]
```diff
@@ -241,14 +241,48 @@ With a domain available, it is time to register an account.
...
+[public key](https://en.wikipedia.org/wiki/Public-key_cryptography), think of it as the "password" used by `mad_hatter@looking_glass`. 
+
+Keys come always come in pairs. It is possible to generate a pair of `public` and `private` keys using one of our
```
Comment (outoftardis) [inline src/guide/bash.md:256]:
```suggestion
Keys always come in pairs. It is possible to generate a pair of `public` and `private` keys using one of our
```

File: src/guide/bash.md [hunk 11]
```diff
@@ -241,14 +241,48 @@ With a domain available, it is time to register an account.
...
+tools, `kagami`. When we've released Iroha, you should be able to install it separately and call like every other program on your system, but for now you can either ask `cargo` to `run` it for you, or build the program and find it in one of the target directories (either `target/debug` or `target/release`).
```
Comment (outoftardis) [inline src/guide/bash.md:257]:
```suggestion
tools, `kagami`. Once Iroha is released, you will be able to install it separately and call it like any other program on your system. For now you can either ask `cargo` to `run` it for you, or build the program and find it in one of the target directories (either `target/debug` or `target/release`).
```

File: src/guide/bash.md [hunk 12]
```diff
@@ -241,14 +241,48 @@ With a domain available, it is time to register an account.
...
+To use `cargo`, run 
+```bash
+$ cargo run --bin kagami  -- crypto
+Public key (multihash): ed012088a5d0cada6edfd4ad8046da64ffce820c16f0e68a21f98df505becd5011e18a
+Private key: 749682bd40695cfc77b5ad599220a80d69b72db5e5078c259635f8e8d8f3559088a5d0cada6edfd4ad8046da64ffce820c16f0e68a21f98df505becd5011e18a
+Digest function: ed25519
+```
+
+::: info 
+You can also use the `-a` flag to specify the algorithm  for key-pair generation and the `-s`
```
Comment (outoftardis) [inline src/guide/bash.md:271]:
```suggestion
You can also use the `-a` flag to specify the algorithm for key-pair generation and the `-s`
```

File: src/guide/bash.md [hunk 13]
```diff
@@ -241,14 +241,48 @@ With a domain available, it is time to register an account.
...
+flag to specify the seed:
```
Comment (outoftardis) [inline src/guide/bash.md:272]:
```suggestion
flag to specify the seed.
```

File: src/guide/bash.md [hunk 14]
```diff
@@ -241,14 +241,48 @@ With a domain available, it is time to register an account.
...
+
+For example, to generate an `ed25519` key (the default algorithm).
```
Comment (outoftardis) [inline src/guide/bash.md:274]: I'd use another (not the default) algorithm as an example here. You can generate the default without the `-a`, it'd make more sense to illustrate the use of this flag with another algorithm

File: src/guide/bash.md [hunk 15]
```diff
@@ -241,14 +241,48 @@ With a domain available, it is time to register an account.
...
+flag to specify the seed:
+
+For example, to generate an `ed25519` key (the default algorithm). 
+
+```bash
+# "cargo run" syntax
+cargo run --bin kagami -- crypto --algorithm ed25519 --seed 42
+Public key (multihash): ed0120ee7a84e8ed0ecc5ae65968ff5465d2bc16f5c6211bc4de0a3f1591e82845f824
+Private key: 790e18703ba023439ee28be38def744ae613cd5e90c643434571b6de80666123ee7a84e8ed0ecc5ae65968ff5465d2bc16f5c6211bc4de0a3f1591e82845f824
+Digest function: ed25519
```
Comment (outoftardis) [inline src/guide/bash.md:281]: you forgot to close the code snippet with ```

File: src/guide/bash.md (hunk 14 above)
Comment (6r1d) [reply src/guide/bash.md:274]: Thanks, while I tried adding an example for `secp256k1`, I've found an LTS bug, so I won't be marking this section as resolved yet.

**Review by appetrosyan — APPROVED** [review]

**Pull request merged**